src/fromager/context.py

```diff
@@ -51,46 +52,50 @@ def __init__(
         # Push items onto the stack as we start to resolve their
         # dependencies so at the end we have a list of items that need to
         # be built in order.
-        self._build_stack = []
-        self._build_requirements = set()
+        self._build_stack: list[typing.Any] = []
+        self._build_requirements: set[tuple[NormalizedName, str]] = set()
 
         # Track requirements we've seen before so we don't resolve the
         # same dependencies over and over and so we can break cycles in
         # the dependency list. The key is the requirements spec, rather
         # than the package, in case we do have multiple rules for the same
         # package.
-        self._seen_requirements = set()
+        self._seen_requirements: set[tuple[NormalizedName, tuple[str, ...], str]] = (
```

Comment:
It feels like `tuple[NormalizedName, tuple[str, ...], str]` should be a [type alias](https://docs.python.org/3/library/typing.html#type-aliases).

Reply:
I left the cleanup of long types as a separate task, in case if these type aliases are common across modules (the last one on the task list in the issue)

Reply:
Are you tracking these planned changes in the #226 ?

Reply:
Yep